src/main/java/org/littleshoot/proxy/impl/DefaultHttpProxyServer.java

```diff
@@ -1010,14 +1027,15 @@ private DefaultHttpProxyServer build() {
                 serverGroup = this.serverGroup;
             }
             else {
-                serverGroup = new ServerGroup(name, clientToProxyAcceptorThreads, clientToProxyWorkerThreads, proxyToServerWorkerThreads);
+                serverGroup = new ServerGroup(name, clientToProxyAcceptorThreads,
+                    clientToProxyWorkerThreads, proxyToServerWorkerThreads, payloadProcessorExecutor);
             }
 
             return new DefaultHttpProxyServer(serverGroup,
                     transportProtocol, determineListenAddress(),
                     sslEngineSource, authenticateSslClients,
                     proxyAuthenticator, chainProxyManager, mitmManagerFactory,
-                    clientToProxyExHandler, proxyToServerExHandler, requestTracer, globalStateHandler,
+                    clientToProxyExHandler, proxyToServerExHandler, requestTracer, globalStateHandler, payloadProcessorExecutor,
```

Comment:
you don't need to pass `payloadProcessorExecutor` again here. it's already embedded into a `serverGroup`.